File: easy/0557_reverse_words_in_a_string_iii/solution.py
class Solution:
    def reverseWords(self, s: str) -> str:
        s_list = list(s)

        i = s.find(" ")
        left = 0

        while i != -1:
            right = i - 1

            while left < right:
                s_list[left], s_list[right] = s_list[right], s_list[left]
                left += 1
                right -= 1

            left = i + 1
            i = s.find(" ", i + 1)

        right = len(s) - 1

        while left < right:
            s_list[left], s_list[right] = s_list[right], s_list[left]
            left += 1
            right -= 1

        return "".join(s_list)

File: easy/0557_reverse_words_in_a_string_iii/solution.cpp
#include <string>

class Solution {
public:
  std::string reverseWords(std::string s) {
    int i = s.find(' ');

    std::string::iterator left = s.begin();
    std::string::iterator right;

    while (i != -1) {
      right = s.begin() + i;
      reverse(left, right);
      left = right + 1;
      i = s.find(' ', i + 1);
    }
    reverse(left, s.end());

    return s;
  }
};
